backend/src/org/commcare/session/SessionFrame.java

```diff
@@ -108,6 +110,28 @@ public StackFrameStep popStep() {
         return recentPop;
     }
 
+    protected boolean rewindToMarkAndSet(String value) {
+        int markIndex = getLatestMarkPosition(steps);
+
+        if (markIndex >= 0) {
+            String markDatumId = steps.get(markIndex).getId();
+            steps = new Vector<>(steps.subList(0, markIndex));
+            steps.addElement(new StackFrameStep(SessionFrame.STATE_DATUM_VAL, markDatumId, value));
```

Comment:
Does it make sense to turn a `mark`/`rewind` pair into `STATE_DATUM_VAL` or `STATE_DATUM_COMPUTED`?